src/multipart/multipart.rs

```diff
@@ -0,0 +1,469 @@
+extern crate uuid;
+
+use std;
+use std::borrow::Cow;
+use std::io::Read;
+use std::ascii::AsciiExt;
+use hyper::mime::Mime;
+
+/// A multipart/form-data request.
+#[derive(Debug)]
+pub struct MultipartRequest {
+    boundary: String,
+    fields: Vec<MultipartField>,
+}
+
+impl MultipartRequest {
+    /// Creates a new MultipartRequest without any content.
+    pub fn new() -> MultipartRequest {
+        MultipartRequest {
+            boundary: format!("{}", uuid::Uuid::new_v4().simple()),
```

Comment:
Currently there is intentionally no way for a user to set the boundary because I did not see a reason why one would set it. I agree that it can be added if a convincing use case is found.